packages/swapkit/cli/src/screens/ConfigInit.tsx

```diff
@@ -0,0 +1,239 @@
+      value: { value: CEI.WALLET_CONNECT_PROJECT_ID, index: 3 },
+      key: 'walletConnect',
+    },
+    {
+      label:
+        itemLabels[CEI.TREZOR_CREDENTIALS] +
+        ': ' +
+        initConfig.trezorManifest?.email +
+        ' ' +
+        initConfig.trezorManifest?.appUrl,
+      value: { value: CEI.TREZOR_CREDENTIALS, index: 4 },
+      key: 'trezor',
+    },
+    {
+      label: itemLabels[CEI.STAGENET] + ': ' + initConfig.stagenet,
+      value: { value: CEI.STAGENET, index: 5 },
+      key: 'stagenet',
+    },
+    {
+      label: itemLabels[CEI.FINISH],
+      value: { value: CEI.FINISH, index: 6 },
+      key: 'finish',
+    },
+  ];
+
+  const staganetItems = [
+    {
+      label: 'true',
+      value: true,
+    },
+    {
+      label: 'false',
+      value: false,
+    },
+  ];
+
+  return (
+    <Box flexDirection="column" rowGap={1}>
+      <Text bold color="magenta">
+        SwapKit Config
+      </Text>
+      {focus === -1 ? (
+        <SelectInput
+          initialIndex={lastSelect}
+          itemComponent={(data) => {
+            const label = data.label.split(':')[0];
+            const value = data.label.split(':')[1]?.trim();
+
+            return (
+              <Text
+                color={data.isSelected ? 'blue' : data.label === 'Finish' ? 'redBright' : 'white'}
+              >
+                {label === 'Finish' ? 'Save Config' : label + ': '}
+                {label === 'Finish' ? (
+                  ''
+                ) : value ? (
+                  <Text color="gray">{value}</Text>
+                ) : (
+                  <Text color="yellow">{'<empty>'}</Text>
+                )}
+              </Text>
+            );
+          }}
+          items={items}
+          onSelect={handleSelect}
+        />
+      ) : focus === CEI.STAGENET ? (
+        <Box>
+          <Text color="green">{itemLabels[focus]}</Text>
+          <Text>: </Text>
+          <SelectInput items={staganetItems} onSelect={staganetOnSelect} />
+        </Box>
+      ) : focus === CEI.TREZOR_CREDENTIALS ? (
+        <Box flexDirection="column">
+          <Text>{itemLabels[focus]}</Text>
+          <Box flexDirection="column" marginTop={1}>
+            <Box>
+              <Text color="green">Email</Text>
+              <Text>: </Text>
+              <TextInput
+                focus={inputFocus === 0 ? true : false}
+                onChange={setQuery}
+                onSubmit={handleEmailSubmit}
+                value={query}
+              />
+            </Box>
+            <Box>
+              <Text color="green">App URL</Text>
+              <Text>: </Text>
+              <TextInput
+                focus={inputFocus === 1 ? true : false}
+                onChange={setQuery2}
+                onSubmit={handleSubmit}
+                value={query2}
+              />
+            </Box>
+          </Box>
+        </Box>
+      ) : focus === CEI.FINISH ? (
+        <>{handleSubmit()}</>

```

Comment:
definitely no go - as it will rerender on every change - should be separate component or tree render